lib/src/rules/pub/secure_pubspec_urls.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/src/lint/pub.dart'; // ignore: implementation_imports
+
+import '../../analyzer.dart';
+
+const _desc = r'Use secure urls in `pubspec.yaml`.';
+
+const _details = r'''
+**DO** Use secure urls in `pubspec.yaml`.
+
+Use `https` instead of `http` an `git:`.
+
+**GOOD:**
+```yaml
+repository: 'https://github.com/dart-lang/example'
+```
+
+**BAD:**
+```dart
+repository: 'http://github.com/dart-lang/example'
+''';
+
+class PubspecNoHttpUrls extends LintRule {
+  PubspecNoHttpUrls()
+      : super(
+            name: 'secure_pubspec_urls',
+            description: _desc,
+            details: _details,
+            group: Group.pub);
+
+  @override
+  PubspecVisitor getPubspecVisitor() => Visitor(this);
+
+  @override
+  LintCode get lintCode => const LintCode(
+      'secure_pubspec_urls', 'The url should only only use secure protocols.',
+      correctionMessage: 'Try using https.');
```

Comment:
Should this read "Try using 'https' or 'git'."? (The protocol should be quoted in either case, but that's a minor nit.)

Reply:
I like this change too.